fedimint-testing/src/gateway.rs

```diff
@@ -86,46 +87,59 @@ impl GatewayTest {
         let client_builder: StandardGatewayClientBuilder =
             StandardGatewayClientBuilder::new(path.clone(), registry, 0);
 
-        let listening_addr = lightning.listening_address();
-        let info = lightning.info().await.unwrap();
-
-        let mut tg = TaskGroup::new();
-        // Create the stream to route HTLCs. We cannot create the Gateway until the
-        // stream to the lightning node has been setup.
-        let (stream, ln_client) = lightning.route_htlcs(&mut tg).await.unwrap();
+        let lightning_builder: Arc<dyn LightningBuilder + Send + Sync> =
+            match Fixtures::is_real_test() {
+                true => Arc::new(RealLightningBuilder {
+                    node_type: lightning.lightning_node_type(),
+                }),
+                false => Arc::new(FakeLightningBuilder {}),
+            };
 
-        let clients = Arc::new(RwLock::new(BTreeMap::new()));
-        let scid_to_federation = Arc::new(RwLock::new(BTreeMap::new()));
+        let gateway_db = Database::new(MemDatabase::new(), decoders.clone());
 
-        // Create gateway with the client created from `route_htlcs`
-        let gateway = Gateway::new(
-            ln_client.clone(),
-            client_builder.clone(),
-            GatewayFee(RoutingFees {
+        let gateway = Gateway::new_with_custom_registry(
+            lightning_builder,
+            client_builder,
+            listen,
+            address.clone(),
+            password.clone(),
+            RoutingFees {
                 base_msat: 0,
                 proportional_millionths: 0,
-            })
-            .0,
-            Database::new(MemDatabase::new(), decoders.clone()),
-            address.clone(),
-            clients.clone(),
-            scid_to_federation.clone(),
-            tg.clone(),
+            },
             num_route_hints,
+            gateway_db,
         )
         .await
-        .unwrap();
-
-        run_webserver(password.clone(), listen, gateway.clone())
+        .expect("Failed to create gateway");
+        gateway
+            .clone()
+            .run()
             .await
-            .expect("Failed to start webserver");
+            .expect("Failed to start gateway");
+
+        // Wait for the gateway to be in the running state
+        let mut gateway_state_iterations = 0;
+        loop {
+            if let GatewayState::Running {
+                lnrpc: _,
+                lightning_public_key: _,
+                lightning_alias: _,
+            } = gateway.state.read().await.clone()
+            {
+                break;
+            }
+
+            if gateway_state_iterations > 9 {
+                panic!("Gateway did not start running after 10 attempts");
+            }
+
+            gateway_state_iterations += 1;
+            sleep(Duration::from_millis(100)).await;
```

Comment:
This is a very low waiting period, it will fail on CI

Reply:
I increased it to 10 seconds, waiting a second each time.